XMonad/Config/Mate.hs

```diff
@@ -51,19 +53,32 @@ mateConfig = desktopConfig
 
 mateKeys (XConfig {modMask = modm}) = M.fromList $
     [ ((modm, xK_p), mateRun)
+    , ((modm, xk_d), unGrab >> matePanel "menu")
     , ((modm .|. shiftMask, xK_q), mateLogout) ]
 
 -- | Launch the "Run Application" dialog.  mate-panel must be running for this
--- to work.
+-- to work.  partial application for keybinding compatibility.
 mateRun :: X ()
-mateRun = withDisplay $ \dpy -> do
+mateRun = matePanel "run"
+
+-- | Launch a panel action. Either the "Run Application" dialog ("run" parameter,
+-- see above) or the "Main Menu" ("menu" parameter).  mate-panel must be running
+-- for this to work.
+matePanel :: String -> X ()
+matePanel action = withDisplay $ \dpy -> do
     rw <- asks theRoot
     mate_panel <- getAtom "_MATE_PANEL_ACTION"
     panel_run  <- getAtom "_MATE_PANEL_ACTION_RUN_DIALOG"
```

Comment:
Since you're passing a string anyway, I would for extensibility pass the appendage to `"_MATE_PANEL_ACTION_"` here instead of hardcoding the two known actions.